py_zipkin/encoding/protobuf/__init__.py

```diff
@@ -0,0 +1,167 @@
+# -*- coding: utf-8 -*-
+import socket
+import struct
+
+from py_zipkin.encoding._types import Kind
+from py_zipkin.util import unsigned_hex_to_signed_int
+try:
+    from py_zipkin.encoding.protobuf import zipkin_pb2
+except ImportError:  # pragma: no cover
+    zipkin_pb2 = None
+
+
+def installed():
+    """Checks whether the protobud library is installed and can be used.
+
+    :return: True if everything's fine, False otherwise
+    :rtype: bool
+    """
+    return zipkin_pb2 is not None
+
+
+def encode_pb_list(pb_spans):
+    """Encode list of protobuf Spans to binary.
+
+    :param pb_spans: list of protobuf Spans.
+    :type pb_spans: list of zipkin_pb2.Span
+    :return: encoded list.
+    :rtype: bytes
+    """
+    pb_list = zipkin_pb2.ListOfSpans()
+    pb_list.spans.extend(pb_spans)
+    return pb_list.SerializeToString()
+
+
+def create_protobuf_span(span):
+    """Converts a py_zipkin Span in a protobuf Span.
+
+    :param span: py_zipkin Span to convert.
+    :type span: py_zipkin.encoding.Span
+    :return: protobuf's Span
+    :rtype: zipkin_pb2.Span
+    """
+
+    # Protobuf's composite types (i.e. Span's local_endpoint) are immutable.
+    # So we can't create a zipkin_pb2.Span here and then set the appropriate
+    # fields since `pb_span.local_endpoint = zipkin_pb2.Endpoint` fails.
+    # Instead we just create the kwargs and pass them in to the Span constructor.
+    pb_kwargs = {}
+
+    pb_kwargs['trace_id'] = _hex_to_bytes(span.trace_id)
+
+    if span.parent_id:
+        pb_kwargs['parent_id'] = _hex_to_bytes(span.parent_id)
+
+    pb_kwargs['id'] = _hex_to_bytes(span.span_id)
+
+    pb_kind = _get_protobuf_kind(span.kind)
+    if pb_kind:
+        pb_kwargs['kind'] = pb_kind
+
+    pb_kwargs['name'] = span.name
```

Comment:
these three fields are nullable btw

Reply:
Also traceId and id? What'd be the use case to emit a span without them?

Reply:
traceId and id are the only required fields in all "v2" formats.. though it is silly to send these alone. The schema language would be a sortof AnyOf thing, where you should send at least something else.. can't really express this.

Reply:
Oh you meant name, timestamp and duration.

Yeah, I'll add the check but there's currently no way in py_zipkin to create a span that doesn't have them.